File: internal/interfaces/polda/polda_service.go
package interfacepolda

import (
	domainpolda "safety-riding/internal/domain/polda"
	"safety-riding/internal/dto"
	"safety-riding/pkg/filter"
)

type ServicePoldaInterface interface {
	Create(req *dto.CreatePoldaAccidentRequest, userID string) error
	GetAll(params filter.BaseParams) ([]domainpolda.PoldaAccident, int64, error)
	GetByID(id string) (*domainpolda.PoldaAccident, error)
	Update(id string, req *dto.UpdatePoldaAccidentRequest, userID string) error
	Delete(id string) error
}

File: frontend/src/services/poldaService.js
import api from './api';

const poldaService = {
  getAll: (params = {}) => api.get('/polda-accidents', { params }),
  getById: (id) => api.get(`/polda-accident/${id}`),
  create: (data) => api.post('/polda-accident', data),
  update: (id, data) => api.put(`/polda-accident/${id}`, data),
  delete: (id) => api.delete(`/polda-accident/${id}`)
};

export default poldaService;
